pkg/server/server.go

```diff
@@ -242,7 +242,9 @@ func (s *Server) Run(ctx context.Context) error {
 		kcpadmissioninitializers.NewKcpInformersInitializer(s.kcpSharedInformerFactory),
 		kcpadmissioninitializers.NewKubeClusterClientInitializer(kubeClusterClient),
 		kcpadmissioninitializers.NewKcpClusterClientInitializer(kcpClusterClient),
-		kcpadmissioninitializers.NewExternalAddressInitializer(genericConfig.ExternalAddress),
+		// The external address pointer is passed as its value may be updated
+		// with the default secure port, when the config is completed.
+		kcpadmissioninitializers.NewExternalAddressInitializer(&genericConfig.ExternalAddress),
```

Comment:
Passing in the pointer means that any admission plugin could potentially mutate genericConfig.ExternalAddress. We presumably wouldn't code that, but it is possible. Do you think we should guard against that?